ui-v2/app/components/modal-dialog.js

```diff
@@ -0,0 +1,87 @@
+import { get, set } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from 'consul-ui/components/dom-buffer';
+import SlotsMixin from 'ember-block-slots';
+import WithResizing from 'consul-ui/mixins/with-resizing';
+
+import templatize from 'consul-ui/utils/templatize';
+const $html = document.documentElement;
+export default Component.extend(SlotsMixin, WithResizing, {
+  // tagName: ''
+  dom: service('dom'),
+  checked: true,
+  height: null,
+  // WithResizing already uses `win`
+  window: null,
+  overflowingClass: 'overflowing',
+  onclose: function() {},
+  onopen: function() {},
+  _open: function(e) {
+    set(this, 'checked', true);
+    if (get(this, 'height') === null) {
+      if (this.element) {
+        const win = [...get(this, 'dom').element('[role="dialog"] > div > div', this.element)][0];
+        const rect = win.getBoundingClientRect();
+        set(this, 'window', win);
+        set(this, 'height', rect.height);
+      }
+    }
+    this.onopen(e);
+  },
+  _close: function(e) {
+    set(this, 'checked', false);
+    const win = get(this, 'window');
+    const overflowing = get(this, 'overflowingClass');
+    if (win.classList.contains(overflowing)) {
+      win.classList.remove(overflowing);
+    }
+    this.onclose(e);
+  },
+  didReceiveAttrs: function() {
+    this._super(...arguments);
+    if (get(this, 'name')) {
+      set(this, 'checked', false);
```

Comment:
`checked` is whether the modal is visible or not. Actually this might be left over from me noodling with the modals, it's doesn't make sense that just because it has a name it means it's not visible. Will recheck this in a bit 👍